### PR TITLE
Add triggers documentation

### PR DESCRIPTION
**Pull request by jpipkin1**
## Purpose of this pull request

This pull request adds the following section:
https://help.sumologic.com/docs/cloud-soar/cloud-soar-integration-framework/#trigger-action-definitions

Issue: [CSOAR-903](https://sumologic.atlassian.net/browse/CSOAR-903)

## Select the type of change:

- [ ] Minor Changes - Typos, formatting, slight revisions, .clabot
- [ ] Update Content - Revisions, updating sections
- [x] New Content - New features, sections, pages, tutorials
- [ ] Site and Tools - Updates, maintenance, dependencies, new packages for the site (Docusaurus, Gatsby, React, etc.)

## PLEASE READ

Following a recent back-end update, all contributors with a local clone or fork of our repository are required to run `yarn install`. This does not apply to [direct page edits](https://help.sumologic.com/docs/contributing/edit-doc/#minor-edits).

- [X] Yes, I've run `yarn install`
- [ ] No, does not apply to me

### CODE REVIEW

File: docs/cloud-soar/cloud-soar-integration-framework.md [hunk 1]
```diff
@@ -514,6 +514,271 @@ Field notes:
...
+
+Triggers allow you to extend Cloud SOAR functionality with custom logic using OIF actions. Triggers are invoked based on specific events that can manually be invoked by Cloud SOAR users, or invoked automatically by interacting with the appropriate endpoint of the API. For example, you can activate a trigger when a button is pressed, a field value is updated in an incident, or when new objects are created. 
+
+#### Define a trigger action
+
+To create a new trigger action, you must define a trigger action YAML file with the value `type` set to `Trigger`:
+
+<img src={useBaseUrl('img/cloud-soar/trigger-code-example.png')} alt="Trigger action" width="800"/>
+
+Depending on the logic that you want to implement in your triggers, specify a list of one or more [hooks](#trigger-hooks) in the trigger YAML file. Each hook represents a manual event or API endpoint that can invoke the trigger. For example, by specifying the hook `updateIncident` inside a trigger, the trigger will fire whenever the field of any incident is updated either manually from the GUI or via the API.
```
Comment (kimsauce) [inline docs/cloud-soar/cloud-soar-integration-framework.md:527]:
```suggestion
Depending on the logic that you want to implement in your triggers, specify a list of one or more [hooks](#trigger-hooks) in the trigger YAML file. Each hook represents a manual event or API endpoint that can invoke the trigger. For example, by specifying the hook `updateIncident` inside a trigger, the trigger will fire whenever the field of any incident is updated either manually from the UI or via the API.
```

File: docs/cloud-soar/cloud-soar-integration-framework.md [hunk 2]
```diff
@@ -514,6 +514,271 @@ Field notes:
...
+
+Triggers function as individual actions, executed in the backend, without the capability to review
```
Comment (kimsauce) [inline docs/cloud-soar/cloud-soar-integration-framework.md:529]: Can you join this with line 530?

File: docs/cloud-soar/cloud-soar-integration-framework.md [hunk 3]
```diff
@@ -514,6 +514,271 @@ Field notes:
...
+
+For other example YAML files, see:
+* [Trigger taskCustomAction definition file (Incident Tools)](/docs/cloud-soar/cloud-soar-integration-framework/#trigger-taskcustomaction-definition-file-incident-tools)
+* [Trigger incidentCustomAction definition file (Incident Tools)](/docs/cloud-soar/cloud-soar-integration-framework/#trigger-incidentcustomaction-definition-file-incident-tools)
+
+#### check_not_null_field
+
+Specifying the YAML field `check_not_null_field : <field ID>` inside a trigger, you can ensure that the trigger is only executed (or displayed in case of custom actions buttons) when a specific incident field is populated.
+
+For example, if we want to display our custom action trigger button only when our custom field w ith
```
Comment (kimsauce) [inline docs/cloud-soar/cloud-soar-integration-framework.md:719]:
```suggestion
For example, if we want to display our custom action trigger button only when our custom field with
```

File: docs/cloud-soar/cloud-soar-integration-framework.md [hunk 4]
```diff
@@ -514,6 +514,271 @@ Field notes:
...
+field ID `opt_15` is populated, inside the YAML file you can set:
```
Comment (kimsauce) [inline docs/cloud-soar/cloud-soar-integration-framework.md:720]: Can you join this with line 719?

File: docs/cloud-soar/cloud-soar-integration-framework.md [hunk 5]
```diff
@@ -514,6 +514,271 @@ Field notes:
...
+
+```json
+check_not_null_field: 'opt_15'
+hook:
+ - incidentCustomAction
+```
+
+#### Define a webhook trigger
+
+Triggers with the hook `webhook` specified can interact with the payload that is posted on your
```
Comment (kimsauce) [inline docs/cloud-soar/cloud-soar-integration-framework.md:730]: Pls join with next line

**Review by kimsauce — CHANGES_REQUESTED** [review]
Minor stuff!

File: docs/cloud-soar/cloud-soar-integration-framework.md [hunk 6]
```diff
@@ -514,6 +514,271 @@ Field notes:
...
+To create a new trigger action, you must define a trigger action YAML file with the value `type` set to `Trigger`:
+
+<img src={useBaseUrl('img/cloud-soar/trigger-code-example.png')} alt="Trigger action" width="800"/>
+
+Depending on the logic that you want to implement in your triggers, specify a list of one or more [hooks](#trigger-hooks) in the trigger YAML file. Each hook represents a manual event or API endpoint that can invoke the trigger. For example, by specifying the hook `updateIncident` inside a trigger, the trigger will fire whenever the field of any incident is updated either manually from the UI or via the API.
+
+Triggers function as individual actions, executed in the backend, without the capability to review
+the execution output in the GUI except for triggers on entities (observables). If a trigger fails, error logs printed on the `stderr` output of the trigger are exported in the audit trail (system log verbosity must be set to `ALL` to review trigger audit logs). Triggers cannot receive manual input, except for [triggers with the `incidentCustomActions` and `taskCustomActions`hooks](#trigger-incidentcustomaction-and-taskcustomaction) that accept a text input.
```
Comment (jpipkin1) [inline docs/cloud-soar/cloud-soar-integration-framework.md:530]:
```suggestion
Triggers function as individual actions, executed in the backend, without the capability to review the execution output in the GUI except for triggers on entities (observables). If a trigger fails, error logs printed on the `stderr` output of the trigger are exported in the audit trail (system log verbosity must be set to `ALL` to review trigger audit logs). Triggers cannot receive manual input, except for [triggers with the `incidentCustomActions` and `taskCustomActions`hooks](#trigger-incidentcustomaction-and-taskcustomaction) that accept a text input.
```

File: docs/cloud-soar/cloud-soar-integration-framework.md [hunk 7]
```diff
@@ -514,6 +514,270 @@ Field notes:
...
+For other example YAML files, see:
+* [Trigger taskCustomAction definition file (Incident Tools)](/docs/cloud-soar/cloud-soar-integration-framework/#trigger-taskcustomaction-definition-file-incident-tools)
+* [Trigger incidentCustomAction definition file (Incident Tools)](/docs/cloud-soar/cloud-soar-integration-framework/#trigger-incidentcustomaction-definition-file-incident-tools)
+
+#### check_not_null_field
+
+Specifying the YAML field `check_not_null_field : <field ID>` inside a trigger, you can ensure that the trigger is only executed (or displayed in case of custom actions buttons) when a specific incident field is populated.
+
+For example, if we want to display our custom action trigger button only when our custom field with
+field ID `opt_15` is populated, inside the YAML file you can set:
```
Comment (jpipkin1) [inline docs/cloud-soar/cloud-soar-integration-framework.md:719]:
```suggestion
For example, if we want to display our custom action trigger button only when our custom field with field ID `opt_15` is populated, inside the YAML file you can set:
```

File: docs/cloud-soar/cloud-soar-integration-framework.md [hunk 8]
```diff
@@ -514,6 +514,269 @@ Field notes:
...
+```json
+check_not_null_field: 'opt_15'
+hook:
+ - incidentCustomAction
+```
+
+#### Define a webhook trigger
+
+Triggers with the hook `webhook` specified can interact with the payload that is posted on your 
+webhook endpoint:
```
Comment (jpipkin1) [inline docs/cloud-soar/cloud-soar-integration-framework.md:729]:
```suggestion
Triggers with the hook `webhook` specified can interact with the payload that is posted on your webhook endpoint:
```

**Comment by jpipkin1** [comment]
Thanks for the review, @kimsauce! Those odd line breaks occurred because I copied and pasted from a PDF. Thanks for catching them.

**Review by kimsauce — APPROVED** [review]

**Pull request merged**